### PR TITLE
[finishes #187354240] setting up docker for backend

### PR DESCRIPTION
**Pull request by JeanIrad**
[finishes #187354240] setting up docker for backend

## Purpose

This PR is designed to setup docker for backend, which is containerizing our development app, for optimized portability
### It has the following services that are currently synced:
       - backend service
       -  database service

## Changes Made

I added a docker file along with docker compose file which allows us to run docker commands interacting with services

## Testing Instructions

I believe there is no test required

## Related Issues

Issues, maybe in how to setup environment variables basically in a .env file

## Checklist

Please review the following checklist and make sure all tasks are complete before submitting:

- [ x] Code follows the project's coding standards
- [x ] Changes are covered by tests
- [ x] Documentation is updated (if applicable)
- [ x] All tests pass

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FROM node:21-alpine3.19
+WORKDIR /app
+RUN apk add --no-cache git
+COPY package*.json /app
+COPY /.git /app/.git
```
Comment (patrickhag) [inline Dockerfile:5]: Aaa kumbe twari twibagiwe gukopiya `.git` file.

**Review by patrickhag — APPROVED** [review]
LGTM: looks good to me

**Review by Favor-star — APPROVED** [review]
It's good!

**Review by Mag-codes — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (JeanIrad) [reply Dockerfile:5]: yeah, hhahahhaha

**Pull request closed without merging**